packages/flutter/lib/src/cassowary/equation_member.dart

```diff
@@ -5,25 +5,64 @@
 import 'expression.dart';
 import 'constraint.dart';
 
-/// Base class for the various parts of cassowary equations.
+/// Represents a member that can be used to construct an expression that may be
+/// used to create a constraint. This is to facilitate the easy creation of
+/// constraints. The use of the operator overloads is completely optional and
+/// is only meant as a convenience. The constraint expressions can be created
+/// by manually creating instance of constraint variables, then terms and
+/// combining those to create expression.
 abstract class EquationMember {
+  /// The representation of this member after it is hoisted to be an
+  /// expression.
   Expression asExpression();
 
+  /// If this member is a constant.
   bool get isConstant;
 
+  /// The current constant value of this member.
   double get value;
 
+  /// Create a constraint by using this member as the left hand side expression
+  /// and the argument as the right hand side expression of a constraint with
+  /// a [Relation.greaterThanOrEqualTo] relationship between the two.
   Constraint operator >=(EquationMember m) => asExpression() >= m;
 
+  /// Create a constraint by using this member as the left hand side expression
+  /// and the argument as the right hand side expression of a constraint with
+  /// a [Relation.lessThanOrEqualTo] relationship between the two.
   Constraint operator <=(EquationMember m) => asExpression() <= m;
 
+  /// Create a constraint by using this member as the left hand side expression
+  /// and the argument as the right hand side expression of a constraint with
+  /// a [Relation.equalTo] relationship between the two.
   Constraint equals(EquationMember m) => asExpression().equals(m);
 
+  /// Create a expression by adding this member with the argument. Both members
+  /// may need to be hoisted to expressions themselves before this can occur.
   Expression operator +(EquationMember m) => asExpression() + m;
 
+  /// Create a expression by subtracting the argument from this member. Both
+  /// members may need to be hoisted to expressions themselves before this can
+  /// occur.
   Expression operator -(EquationMember m) => asExpression() - m;
 
+  /// Create a expression by multiplying this member with the argument. Both
+  /// members may need to be hoisted to expressions themselves before this can
+  /// occur.
+  ///
+  /// Warning: This operation may throw an exception if the resulting expression
+  /// is no longer linear. This is because a non-linear expression may not
+  /// be used to create a constraint. At least one of the expression members
+  /// must be evaluatable to be a constant.
```

Comment:
s/must be evaluatable to be a constant/must evaluate to a constant/ or
s/must be evaluatable to be a constant/must be a constant/